src/expressions/include/antares/expressions/iterators/pre-order.h

```diff
@@ -3,7 +3,7 @@
 #include <stack>
 #include <vector>
 
-namespace Antares::Solver::Nodes
```

Comment:
Why is this file named **pre-order.h** whereas it contains declaration of class **AST** ?
**AST** seems to be the main topic here, at least the feature we should highlight.
Class **ASTPreOrderIterator** is an implementation detail.
Rename into **ast.h** ?

Reply:
@a-zakir opinion ?